packages/rum/src/domain/segmentCollection/segmentCollection.ts

```diff
@@ -95,16 +97,16 @@ export function doStartSegmentCollection(
     }
   )
 
-  function flushSegment(nextSegmentCreationReason?: CreationReason) {
+  function flushSegment(flushReason: FlushReason) {
     if (state.status === SegmentCollectionStatus.SegmentPending) {
-      state.segment.flush()
+      state.segment.flush(flushReason)
       clearTimeout(state.expirationTimeoutId)
     }
 
-    if (nextSegmentCreationReason) {
+    if (flushReason !== 'stop') {
```

Comment:
would be good to have a test for the case of `SegmentCollectionStatus.Stopped`

Reply:
There is a test to check that we correctly frush on stop: https://github.com/DataDog/browser-sdk/blob/fecae5fb3fd956930fa92fbcdd8c1b5157d45508/packages/rum/src/domain/segmentCollection/segmentCollection.spec.ts#L319

Do you have something else in mind?